vulnerabilities/importers/npm.py

```diff
@@ -8,191 +8,125 @@
 
 
 class NpmImporter(GitImporter):
-    def __enter__(self):
-        super(NpmImporter, self).__enter__()
-        if not getattr(self, "_added_files", None):
-            self._added_files, self._updated_files = self.file_changes(
-                recursive=True, file_ext="json", subdir="./vuln/npm"
-            )
-
-        self._versions = NpmVersionAPI()
-        self.set_api(self.collect_packages())
-
-    def updated_advisories(self) -> Set[AdvisoryData]:
-        files = self._updated_files.union(self._added_files)
-        advisories = []
-        for f in files:
-            processed_data = self.process_file(f)
-            if processed_data:
-                advisories.extend(processed_data)
-        return self.batch_advisories(advisories)
-
-    def set_api(self, packages):
-        asyncio.run(self._versions.load_api(packages))
-
-    def collect_packages(self):
-        packages = set()
-        files = self._updated_files.union(self._added_files)
-        for f in files:
-            data = load_json(f)
-            packages.add(data["module_name"].strip())
-
-        return packages
-
-    @property
-    def versions(self):  # quick hack to make it patchable
-        return self._versions
-
-    def process_file(self, file) -> List[AdvisoryData]:
-
-        record = load_json(file)
-        advisories = []
-        package_name = record["module_name"].strip()
-
-        publish_date = parse(record["updated_at"])
-        publish_date = publish_date.replace(tzinfo=pytz.UTC)
-
-        all_versions = self.versions.get(package_name, until=publish_date).valid_versions
-        aff_range = record.get("vulnerable_versions")
-        if not aff_range:
-            aff_range = ""
-        fixed_range = record.get("patched_versions")
-        if not fixed_range:
-            fixed_range = ""
-
-        if aff_range == "*" or fixed_range == "*":
-            return []
-
-        impacted_versions, resolved_versions = categorize_versions(
-            all_versions, aff_range, fixed_range
+    license_url = "https://github.com/nodejs/security-wg/blob/main/LICENSE.md"
+    spdx_license_expression = "MIT"
+
+    def __init__(self):
+        super().__init__(repo_url="git+https://github.com/nodejs/security-wg.git")
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        self.clone()
+        path = Path(self.vcs_response.dest_dir)
+
+        glob = "vuln/npm/**/*.json"  # subdir="vuln/npm"
+        files = (p for p in path.glob(glob) if p.is_file())
+        for file in files:
+            print(file)
+            record = load_json(file)
+            yield parse_advisory_data(record)
+
+
+def parse_advisory_data(record) -> Optional[AdvisoryData]:
+    cves = record.get("cves") or []
+    overview = record.get("overview", "")
+    package_name = record["module_name"].strip()
+
+    publish_date = parse(record["updated_at"])
+    publish_date = publish_date.replace(tzinfo=pytz.UTC)
+
+    pkg_manager_api = NpmVersionAPI()
```

Comment:
VersionAPI is needed to be handled at improver side, check github and gitlab importer for same